Makefile

```diff
@@ -757,6 +757,13 @@ pkgs: RESCAN_DEPS=
 pkgs: build-tools $(PKGS)
 	@echo Done building packages
 
+pkg/external-boot-image: eve-external-boot-image
```

Comment:
@zedi-pramodh , is there any particular reason to create this intermediary external-boot-image.tar tarball image? Because from this PR I don't understand why this file is needed.... IMO you could let everything inside the containers, like this:

1. You've created the pkg/external-boot-image container to extract from eve-kernel the kernel and initrd images
2. You could create the tarball directly from the Dockerfile (of pkg/external-boot-image) and put it inside this container
3. In the Dockerfile of pkg/kube you can just import the generated tarball directly from the pkg/external-boot-image container

In this way you can avoid creating two new targets in the Makefile that will need to be maintained and letting the tarball generated outside the build system, even though you still need the tarball externally, you can export containers from linuxkit cache using the `make export-cache IMAGE=.... OUTFILE=....` target.... but as mentioned, I don't know the whole context and if there is a particular reason to keep this intermediary file....

Reply:
Thinking about what @eriknordmark commented in this PR, another option would be generate the tarball during runtime, since you will always have both files that you need: kernel + initrd... this approach would:

1. Avoid the creation of the new container
2. Avoid the increasing of the rootfs
3. Avoid problems with different architectures/kernel version, since you will always pick the working (running) kernel + initrd

Drawbacks: You will need to create the tarball (at least once) during runtime, which I think is not a problem at all...

Reply:
@rene I am using new container to generate a tarball which gets shipped in kubevirt images only.

The tarball is a docker image tarball (not just collection of kernel and initrd binaries), since we do not have docker in the runtime of the eve, we have to do at build time.

But I liked your comment on eliminating extra Makefile target to achieve the same result.  Would like to learn more on that.

Reply:
@zedi-pramodh , as I said, you can use the cache-export from our Makefile instead of call docker directly and export the external-boot-image directly in the pkg/kube target, something like this:

```diff
diff --git a/Makefile b/Makefile
index 9e204372a..02a9f849c 100644
--- a/Makefile
+++ b/Makefile
@@ -362,7 +362,7 @@ PKGS=pkg/alpine $(PKGS_$(ZARCH))
 
 # these are the packages that, when built, also need to be loaded into docker
 # if you need a pkg to be loaded into docker, in addition to the lkt cache, add it here
-PKGS_DOCKER_LOAD=mkconf mkimage-iso-efi mkimage-raw-efi mkverification-raw-efi mkrootfs-ext4 mkrootfs-squash external-boot-image
+PKGS_DOCKER_LOAD=mkconf mkimage-iso-efi mkimage-raw-efi mkverification-raw-efi mkrootfs-ext4 mkrootfs-squash
 # these packages should exists for HOSTARCH as well as for ZARCH
 # alpine-base, alpine and cross-compilers are dependencies for others
 PKGS_HOSTARCH=alpine-base alpine cross-compilers $(PKGS_DOCKER_LOAD)
@@ -757,12 +757,9 @@ pkgs: RESCAN_DEPS=
 pkgs: build-tools $(PKGS)
        @echo Done building packages
 
-pkg/external-boot-image: eve-external-boot-image
-       docker tag $(shell $(LINUXKIT) pkg show-tag pkg/external-boot-image) lfedge/eve-external-boot-image:latest
-       docker save -o pkg/kube/external-boot-image.tar lfedge/eve-external-boot-image:latest
-       $(QUIET): $@: Succeeded
-pkg/kube: pkg/external-boot-image eve-kube
-       rm -f pkg/kube/external-boot-image.tar
+pkg/kube: eve-external-boot-image
+       $(MAKE) cache-export IMAGE=$(shell $(LINUXKIT) pkg show-tag pkg/external-boot-image) OUTFILE=pkg/kube/external-boot-image.tar && \
+       $(MAKE) eve-kube && rm -f pkg/kube/external-boot-image.tar
        $(QUIET): $@: Succeeded
 pkg/pillar: pkg/dnsmasq pkg/gpt-tools pkg/dom0-ztools eve-pillar
        $(QUIET): $@: Succeeded
```

Reply:
Thanks, I will try this out and see how it goes.

Reply:
This worked but I cannot pass in the env variables to linuxkit build.